### DISCUSSION TITLE
advice on how to SSL / https terria map

### DISCUSSION THREAD

**Discussion by hardreddata** [TerriaJS/terriajs#7235] [Q&A]
I found this little nugget on the terriajs-server github 

_It can be run in HTTPS mode, although there are better ways of doing that in production._

Advice invited.

Technically we still run v7 so maybe this will be the push we need to migrate.

Any advice which involves IIS on Windows would be great. 

Many thanks.

**Comment by na9da** [comment]
Theoretically you can configure terriajs-server to handle HTTPS like in [this example](https://github.com/TerriaJS/terriajs-server/blob/2c48621ffdf36a7a764bfb981f75820781e383b9/serverconfig.json.example#L176-L182). 

But it might be better to make a gateway server handle it as it might be more performant and safe(?). In this setup terriajs-server works in HTTP mode and is oblivious to the fact that there is an HTTPS in front. I think there should be no difference between v7 and v8 in this regard.

Mozilla has an [SSL configuration generator](https://ssl-config.mozilla.org/) for various servers, but I don't see IIS in that list. Sorry, I don't have much experience with IIS setup, but you might want to look for setting up a reverse proxy mode - [something like this](https://dev.to/petereysermans/hosting-a-node-js-application-on-windows-with-iis-as-reverse-proxy-397b).

**Comment by hardreddata** [comment]
Thanks. 

Presently using Squid for the http reverse proxy but running on Windows so IIS possibly makes more sense.
